src/app/product/productList/page.tsx

```diff
@@ -0,0 +1,148 @@
+"use client";
+import { Table,TableBody,TableCell,TableHead,TableHeader,TableRow} from "../../../components/ui/table";
+import {AlertDialog,AlertDialogAction,AlertDialogCancel,AlertDialogContent,AlertDialogDescription,AlertDialogFooter,AlertDialogHeader,AlertDialogTitle} from "../../../components/ui/alert-dialog";
+import { Typography,Stack,IconButton,CircularProgress,Box} from "@mui/material";
+import { useEffect, useState } from "react";
+import EditIcon from "@mui/icons-material/Edit";
+import DeleteIcon from "@mui/icons-material/Delete";
+import { Button } from "../../../components/ui/button";
+import { useRouter } from "next/navigation";
+import useRequests from "../../service/request";
+import axios from "axios";
+
+interface Product {
+  id:string,
+  name: string;
+  description: string;
+  price: number | null;
+}
+
+export default function ProductTable() {
+  const endpoints = useRequests();
+  const [loading, setLoading] = useState(true);
+  const [product, setProduct] = useState<Product[]>([]);
+  const [selectedProduct, setSelectedProduct] = useState<Product | null>(null);
+  const [isDeleteDialogOpen, setIsDeleteDialogOpen] = useState(false);
+  const router = useRouter();
+
+  useEffect(() => {
+    getProduct();
```

Comment:
This is more of react way of fetching on client, look through best practices in next js